apps/wallet-mobile/src/features/AddWallet/useCases/CreateWallet/VerifyRecoveryPhrase.tsx

```diff
@@ -0,0 +1,435 @@
+import {useTheme} from '@yoroi/theme'
+import * as React from 'react'
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native'
+import {ScrollView} from 'react-native-gesture-handler'
+import LinearGradient from 'react-native-linear-gradient'
+import Animated, {FadeIn, FadeOut, Layout} from 'react-native-reanimated'
+import {SafeAreaView} from 'react-native-safe-area-context'
+
+import {Button, StatusBar} from '../../../../components'
+import {Space} from '../../../../components/Space/Space'
+import {mockAddWallet} from '../../common/mocks'
+import {StepperProgress} from '../../common/StepperProgress/StepperProgress'
+import {useStrings} from '../../common/useStrings'
+import AlertIcon from '../../illustrations/AlertIcon'
+import Check2Icon from '../../illustrations/Check2Icon'
+
+export const VerifyRecoveryPhrase = () => {
+  const {styles} = useStyles()
+
+  const strings = useStrings()
+
+  const mnemonic = mockAddWallet.mnemonic
+
+  const mnemonicEntries: Array<Entry> = mnemonic
+    .split(' ')
+    .sort()
+    .map((word: string, id: number) => ({word, id}))
+
+  const mnemonicDefault: Array<Entry> = mnemonic.split(' ').map((word: string, id: number) => ({word, id}))
+
+  const [userEntries, setUserEntries] = React.useState<Array<Entry>>([])
```

Comment:
I'd prefer to build as a map, and replace by the last empty index, it will support more features like dropping out of the order word, and it would be simpler